tests/unittests/config/test_cc_raspberry_pi.py

```diff
@@ -57,20 +57,6 @@ def test_handle_configure_serial_interface_bool(self, m_ispi5, m_serial):
 
 
 class TestRaspberryPiMethods:
-    @mock.patch("cloudinit.subp.subp")
-    def test_configure_rpi_connect_enable(self, m_subp):
```

Comment:
There now appears to be no unittest coverage for `configure_usb_gadget`, we don't want to reduce test coverage as we add new functionality if possible. I think If we are able to rewrite the logic to actually invoke /usr/bin/rpi-usb-gadget command directly with subp.subp, we should be able to unittest that functionality to ensure it's called from cloud-init.

Reply:
Previously, this wasn’t possible, which is why I opted for the earlier approach. Since some things have changed, I’ve now moved it into the config stage and execute it directly, and I also shifted the restart from our RPi OS powerstate change hook to the same mechanism used for other config.txt changes.

I agree that we shouldn’t reduce test coverage when adding new functionality. I now added test coverage for the changes.